sql/core/src/main/java/org/apache/spark/sql/sources/v2/DataSourceOptions.java

```diff
@@ -92,8 +94,8 @@ public DataSourceOptions(Map<String, String> originalMap) {
     }
   }
 
-  public Map<String, String> asMap() {
-    return new HashMap<>(keyLowerCasedMap);
+  public CaseInsensitiveMap<String> asMap() {
```

Comment:
I think there was a discussion that returning `CaseInsensitiveMap` was a bad design before (IIRC). I think that's the reason why it return a regular map instead. Why don't we just clearly document what it returns? I think this can be a conservative compromise as well.

Reply:
Is there any link for the discussion? I didn't find in #19925.

Reply:
I think I talked with @cloud-fan and @rxin. I might remember this wrongly. `CaseInsensitiveMap` is an internal class from `catalyst`. We shouldn't expose it like this anyway.